### DISCUSSION TITLE
customizing theme with nested names works for colors but not others?

### DISCUSSION THREAD

**Discussion by stevenjr1988** [tailwindlabs/tailwindcss#9996] [Help]
so when customizing, I noticed I can do something like this with the colors (below), and it works!
```
theme: {
  extend: {
    colors: {
      some: {
        crazy: {
          stupid: {
            nonsense: {
              unneeded: {
                long: {
                  name: {
                    with: {
                      a: {
                        number: {
                          of: {
                            nested: {
                              children: {
                                1: 'rgb(250, 50, 100)',
                                2: 'rgb(50, 250, 100)',
                                3: 'rgb(50, 100, 250)',
                              }
                            }
                          }
                        }
                      }
                    }
                  }
                }
              }
            }
          }
        }
      }
    }
  }
}
```

however, with boxShadow, if I even try to do more then 1 nested child like below, tailwind seams to break. without error.

```
theme: {
  extend: {
      boxShadow: {
        mcck: {
          'sm' : '0 35px 60px -15px rgba(255, 0, 0, 0.3)',
        }
      }
  }
}
```

any ideas?

**Comment by wongjn** [comment]
`boxShadow` values can be defined in `theme.boxShadow` or `theme.extend.boxShadow`, not in `theme.extend.colors.boxShadow`. Additionally, the core plugin for box shadow utilities expect only a single depth object. For colors, there is special handling for variable object depth.

**Reply by stevenjr1988** [reply]
sorry, that was my mistake I originally edited the post, and copy/pated, the theme.extend.colors, and did not edit the 2nd part to
theme.extend.boxShadow

**Reply by wongjn** [reply]
OK, my other points still stand to answer your query:
> […] the core plugin for box shadow utilities expect only a single depth object. For colors, there is special handling for variable object depth.

**Reply by adamwathan** [reply]
Yeah as @wongjn mentioned only colors have support for that magic flattening stuff.

I would recommend just doing this:

```js
theme: {
  extend: {
    boxShadow: {
      'mcck-sm': '0 35px 60px -15px rgba(255, 0, 0, 0.3)',
    }
  }
}
```
